pkg/apis/common/common.go

```diff
@@ -136,9 +136,9 @@ type SASLConfig struct {
 	Mechanism string `json:"mechanism,omitempty" protobuf:"bytes,1,opt,name=mechanism"`
 	// User is the authentication identity (authcid) to present for
 	// SASL/PLAIN or SASL/SCRAM authentication
-	User *corev1.SecretKeySelector `json:"user,omitempty" protobuf:"bytes,2,opt,name=user"`
+	UserSecret *corev1.SecretKeySelector `json:"user,omitempty" protobuf:"bytes,2,opt,name=user"`
```

Comment:
1. If `json: ""user,xxxxx,name=user` is not changed, the spec is not changed.
2. A good practice would be adding 2 additional properties, and mark these existing 2 as `Deprecated`, and state they will be unsupported after 2 major releases (v1.6).  Take line 120-122 as an example.

Reply:
Doh! Missed #1, good catch, I was considering #2, but since its unreleased I wasn't sure if the cruft would be needed

Reply:
Oh... I forgot it's not released yet... then ignore the second one.